### PR TITLE
new diopirt api for storage/deviceindex

### PR DESCRIPTION
**Pull request by wyz5864**
## Motivation and Context

为了解决 https://github.com/DeepLink-org/DIPU/issues/334

dipu添加新api支持：https://github.com/DeepLink-org/DIPU/pull/417

## Description

添加了几个diopirt的api来获取storag的ptr\nbytes，tensor的offset\device_index

## Checklist
**Before PR**:

- [ ] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: diopi_test/diopi_stub/csrc/litert.cpp [hunk 1]
```diff
@@ -245,6 +245,26 @@ DIOPI_RT_API diopiError_t diopiGetTensorElemSize(diopiConstTensorHandle_t th, in
     return diopiSuccess;
 }
 
+DIOPI_RT_API diopiError_t diopiGetTensorStoragePtr(diopiConstTensorHandle_t th, void** pStoragePtr) {
+    *pStoragePtr = const_cast<void*>(th->data());
+    return diopiSuccess;
+}
+
+DIOPI_RT_API diopiError_t diopiGetTensorStorageOffset(diopiConstTensorHandle_t th, int64_t* pOffset) {
+    *pOffset = 0;
```
Comment (fandaoyi) [inline diopi_test/diopi_stub/csrc/litert.cpp:254]: 所以 diopi test 里现在没有view的概念？

File: diopi_test/diopi_stub/csrc/litert.cpp [hunk 2]
```diff
@@ -245,6 +245,26 @@ DIOPI_RT_API diopiError_t diopiGetTensorElemSize(diopiConstTensorHandle_t th, in
...
+    return diopiSuccess;
+}
+
+DIOPI_RT_API diopiError_t diopiGetTensorStorageNbytes(diopiConstTensorHandle_t th, size_t* pNbytes) {
+    *pNbytes = th->nbytes();
+    return diopiSuccess;
+}
+
+DIOPI_RT_API diopiError_t diopiGetTensorDeviceIndex(diopiConstTensorHandle_t th, diopiDeviceIndex_t* pDevIndex) {
+    *pDevIndex = 0;
```
Comment (fandaoyi) [inline diopi_test/diopi_stub/csrc/litert.cpp:264]: diopi 现在不支持多卡是吧？ 所以diopi 只能跑在 卡 0吗？ 要不加个注释说明下。

File: diopi_test/diopi_stub/csrc/litert.cpp (hunk 1 above)
Comment (yangbofun) [reply diopi_test/diopi_stub/csrc/litert.cpp:254]: 目前diopi test没有这种测例生成机制，当添加时，我们会在修改这里。

File: diopi_test/diopi_stub/csrc/litert.cpp (hunk 2 above)
Comment (yangbofun) [reply diopi_test/diopi_stub/csrc/litert.cpp:264]: > diopi test现在不支持多卡是吧？ 所以diopi test 只能跑在 卡 0吗？ 要不加个注释说明下。

diopi test只测 数学意义上的算子实现，目前没有通讯接口，不测多卡。

File: diopi_test/diopi_stub/csrc/litert.cpp (hunk 1 above)
Comment (yangbofun) [reply diopi_test/diopi_stub/csrc/litert.cpp:254]: > 所以 diopi test 里现在没有view的概念？

目前diopi test没有这种测例生成机制，当添加时，我们会在修改这里。

**Review by yangbofun — APPROVED** [review]

File: diopi_test/diopi_stub/csrc/litert.cpp (hunk 1 above)
Comment (wyz5864) [reply diopi_test/diopi_stub/csrc/litert.cpp:254]: 1、从现状上看：diopiTensor这个类里没有offset\deviceindex相关的定义 也许有view（比如测例config中带stride的实现？）但是应该没有slice相关的操作（offset永远是0） https://github.com/DeepLink-org/DIOPI/blob/6717aaf27007390968846e87f96ed0be2b8b3070/diopi_test/diopi_stub/include/litert.hpp#L69
2、从逻辑上说：目前storage\offset\deviceindex的语义完全由at::Tensor提供，diopiTensorHandle_t 的协议里是否提供有待商榷

**Review by fandaoyi — APPROVED** [review]

**Pull request merged**